### PR TITLE
Remove unnecessary features to decrease maintenance/complexity overhead. + Cleanup & more repo usage.

### PR DESCRIPTION
**Pull request by cmyui**
# Describe your changes

This PR removes a couple of features:
- Support for unix sockets and related checks
- Insurance that the running python version is (>=3.11)
- Support for deprecated `SERVER_ADDR` and `SERVER_PORT` config vars - now only `APP_x` variants exist
- Exception hooks to catch some cases and display them in more simple light
- Validations to INET addresses
- Debugging hooks under `_testing.runtime`
- CLI handling using argparse for `main.py` (was only used for `./main.py -V`)

All of these seem like unnecessary complexity requiring extra maintenance overhead with low value delivered.

It also migrates more direct queries to SQL to instead use `players_repo`, removes usage of `__all__` in modules, and more general cleanup.

## Checklist
- [x] I've manually tested my code

### CODE REVIEW

**Comment by cmyui** [comment]
This is a terrible PR - sorry - moving fast today

File: app/commands.py [hunk 1]
```diff
@@ -1258,7 +1259,7 @@ async def server(ctx: Context) -> str | None:
 
     return "\n".join(
         (
-            f"{build_str} | uptime: {seconds_readable(uptime)}",
+            f"{build_str} | uptime: {timedelta(seconds=uptime)}",
```
Comment (cmyui) [inline app/commands.py:1262]: today i learned: str(timedelta()) does exactly what i want

File: app/objects/player.py [hunk 2]
```diff
@@ -238,7 +238,6 @@ def __init__(
 
         self.id = id
         self.name = name
-        self.safe_name = self.make_safe(self.name)
```
Comment (cmyui) [inline app/objects/player.py:241]: Moved to virtual property below

File: app/settings.py [hunk 3]
```diff
@@ -2,30 +2,16 @@
...
-APP_PORT = None
-_app_port = support_deprecated_vars(
-    new_name="APP_PORT",
-    deprecated_name="SERVER_PORT",
-    until=date(2024, 1, 1),
-    allow_empty_string=True,
-)
-if _app_port:
-    APP_PORT = int(_app_port)
+APP_HOST = os.environ["APP_HOST"]
```
Comment (NiceAesth) [inline app/settings.py:13]: update docker configs as well

Comment (cmyui) [reply app/settings.py:13]: Should be good I think:

image

We previously deprecated SERVER_HOST and SERVER_PORT back in Sep 2023; this just fully removes support for the already-deprecated vars.

Comment (NiceAesth) [reply app/settings.py:13]: https://github.com/osuAkatsuki/bancho.py/blob/4310cf0769f903c3c9d81dc9cb109986530405a1/docker-compose.yml#L89

Comment (NiceAesth) [reply app/settings.py:13]: (same thing in test yml)

Comment (cmyui) [reply app/settings.py:13]: Oo ty didn't notice

Comment (cmyui) [reply app/settings.py:13]: Aha, in our deprecation msg we typo'd SERVER_ADDR to SERVER_HOST

**Review by NiceAesth — DISMISSED** [review]
O K TO me (albeit very scary)

**Comment by TheFantasticLoki** [comment]
My only gripe is the no internet check, can be very useful in certain situations. otherwise good PR

**Comment by cmyui** [comment]
> My only gripe is the no internet check, can be very useful in certain situations. otherwise good PR

👍🏻 I've re-added it

**Review by NiceAesth — APPROVED** [review]
ok

**Review by NiceAesth — APPROVED** [review]

**Pull request merged**